src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp

```diff
@@ -496,6 +496,13 @@ void ReplicatedMergeTreeSinkImpl<async_insert>::consume(Chunk chunk)
         ));
     }
 
+    if (streams > 0 && streams <= max_insert_delayed_streams_for_parallel_write)
```

Comment:
Sorry that I still don't understand why this will fix the bug. Could you please elaborate (and also comments in the code)?

Reply:
The bug appears when the condition in line 476 does not happen, and the code in 479-481 is not executed, which provokes the data not moving to the MV.

This extra lines are just make sure those functions are executed only when the condition in 476 is not executed.

Reply:
but line 507, 508, 515 does the same thing as line 479-481, why does not this part of code move data to MV?

Reply:
You are right, I have been doing some tests and for some reason what makes it work is recreating delayed_chunk after calling finishDelayedChunk. I have updated the PR with the change. But I honestly don't understand why this change makes it work